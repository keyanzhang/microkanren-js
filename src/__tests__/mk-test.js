const {
  append,
  callWithFresh,
  callWithInitialState,
} = require('../mk');

const {
  list,
  toString,
} = require('../lib');

it('should match snapshots', () => {
  const result1 = toString(callWithInitialState(
    false,
    callWithFresh((q) => append(
      list('t', 'u', 'v'),
      list('w', 'x'),
      q
    ))
  ));

  expect(result1).toMatchSnapshot();

  const result2 = toString(callWithInitialState(
    false,
    callWithFresh(
      (q) => callWithFresh(
        (p) => append(
          q,
          p,
          list('t', 'u', 'v', 'w', 'x')
        )
      )
    )
  ));

  expect(result2).toMatchSnapshot();
});
