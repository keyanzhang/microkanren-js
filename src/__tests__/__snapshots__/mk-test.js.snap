exports[`test should match snapshots 1`] = `"((((9 w x) (6 . (7 . 9)) (8) (7 . v) (3 . (4 . 6)) (5 v) (4 . u) (0 . (1 . 3)) (2 u v) (1 . t)) . 10))"`;

exports[`test should match snapshots 2`] = `"((((1 t u v w x) (0)) . 2) (((1 u v w x) (3) (4 u v w x) (2 . t) (0 . (2 . 3))) . 5) (((1 v w x) (6) (7 v w x) (5 . u) (3 . (5 . 6)) (4 u v w x) (2 . t) (0 . (2 . 3))) . 8) (((1 w x) (9) (10 w x) (8 . v) (6 . (8 . 9)) (7 v w x) (5 . u) (3 . (5 . 6)) (4 u v w x) (2 . t) (0 . (2 . 3))) . 11) (((1 x) (12) (13 x) (11 . w) (9 . (11 . 12)) (10 w x) (8 . v) (6 . (8 . 9)) (7 v w x) (5 . u) (3 . (5 . 6)) (4 u v w x) (2 . t) (0 . (2 . 3))) . 14) (((1) (15) (16) (14 . x) (12 . (14 . 15)) (13 x) (11 . w) (9 . (11 . 12)) (10 w x) (8 . v) (6 . (8 . 9)) (7 v w x) (5 . u) (3 . (5 . 6)) (4 u v w x) (2 . t) (0 . (2 . 3))) . 17))"`;
